### PR TITLE
feat(completion-progress): add alphabetical sorting

### PR DESCRIPTION
**Pull request by wescopeland**
This PR adds two new sort orders to the Completion Progress page:
Game Title: A - Z
Game Title: Z - A

All the sort options have also been relabeled to hopefully improve the intuitiveness of the sort select control.

Screenshot 2023-11-18 at 10 23 26 AM

### CODE REVIEW

File: resources/views/platform/completion-progress-page.blade.php [hunk 1]
```diff
@@ -58,7 +58,7 @@
             @endif
 
             @if ($isFiltering || $selectedSortOrder !== 'unlock_date')
-                <a href="{{ route('user.completion-progress', $targetUsername) }}" class="btn flex items-center gap-x-0.5">
+                <a href="{{ route('user.completion-progress', $targetUsername) }}" class="btn flex items-center gap-x-0.5 transition lg:active:scale-95">
```
Comment (wescopeland) [inline resources/views/platform/completion-progress-page.blade.php:61]: Minor - adds the press effect to this reset filters link button.

File: app/Platform/Controllers/PlayerCompletionProgressController.php [hunk 2]
```diff
@@ -131,6 +131,18 @@ private function useSortedList(array $filteredAndJoinedGamesList, string $sortOr
             });
         }
 
+        if ($sortOrder === 'game_title') {
+            usort($filteredAndJoinedGamesList, function ($a, $b) {
+                return strcmp($a['Title'], $b['Title']);
```
Comment (Jamiras) [inline app/Platform/Controllers/PlayerCompletionProgressController.php:136]: Not sure if it matters, but Hacks and Homebrew are sorted after normal games.
image

Developer sets has some code that generates a sort title and uses that for sorting:
https://github.com/RetroAchievements/RAWeb/blob/f5105523a1fe8b6eb78114a8130f23d77c7b4aee/app/Platform/Controllers/DeveloperSetsController.php#L134-L138

image

Comment (wescopeland) [reply app/Platform/Controllers/PlayerCompletionProgressController.php:136]: I think it matters, will take a look.

Comment (wescopeland) [reply app/Platform/Controllers/PlayerCompletionProgressController.php]: Thanks, resolved in https://github.com/RetroAchievements/RAWeb/pull/2039/commits/e86335d7a95ad48dbcc73d262bb1643cd79d408c.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**